src/code/z_room.c

```diff
@@ -580,32 +587,52 @@ u32 func_80096FE8(PlayState* play, RoomContext* roomCtx) {
 
             PRINTF("DOOR%d=<%d> ROOM1=<%d, %d> ROOM2=<%d, %d>\n", j, cumulRoomSize, frontRoom, frontRoomSize, backRoom,
                    backRoomSize);
-            if (maxRoomSize < cumulRoomSize) {
-                maxRoomSize = cumulRoomSize;
+            if (roomBufferSize < cumulRoomSize) {
+                roomBufferSize = cumulRoomSize;
             }
             transitionActor++;
         }
     }
 
     PRINTF(VT_FGCOL(YELLOW));
-    PRINTF(T("部屋バッファサイズ=%08x(%5.1fK)\n", "Room buffer size=%08x(%5.1fK)\n"), maxRoomSize,
-           maxRoomSize / 1024.0f);
-    roomCtx->bufPtrs[0] = GAME_STATE_ALLOC(&play->state, maxRoomSize, "../z_room.c", 946);
+    PRINTF(T("部屋バッファサイズ=%08x(%5.1fK)\n", "Room buffer size=%08x(%5.1fK)\n"), roomBufferSize,
+           roomBufferSize / 1024.0f);
+    roomCtx->bufPtrs[0] = GAME_STATE_ALLOC(&play->state, roomBufferSize, "../z_room.c", 946);
     PRINTF(T("部屋バッファ開始ポインタ=%08x\n", "Room buffer initial pointer=%08x\n"), roomCtx->bufPtrs[0]);
-    roomCtx->bufPtrs[1] = (void*)((uintptr_t)roomCtx->bufPtrs[0] + maxRoomSize);
+    roomCtx->bufPtrs[1] = (void*)((uintptr_t)roomCtx->bufPtrs[0] + roomBufferSize);
     PRINTF(T("部屋バッファ終了ポインタ=%08x\n", "Room buffer end pointer=%08x\n"), roomCtx->bufPtrs[1]);
     PRINTF(VT_RST);
-    roomCtx->unk_30 = 0;
+    roomCtx->activeBufPage = 0;
     roomCtx->status = 0;
 
     frontRoom = gSaveContext.respawnFlag > 0 ? ((void)0, gSaveContext.respawn[gSaveContext.respawnFlag - 1].roomIndex)
                                              : play->spawnList[play->spawn].room;
-    func_8009728C(play, roomCtx, frontRoom);
 
-    return maxRoomSize;
+    // Load into a room for the first time.
+    // Since curRoom was initialized to (room -1, segment NULL) in Play_InitScene, the previous room
```

Comment:
I dont think parens make sense here, can prob just phrase as 
`Since curRoom was initialized to room -1 and a NULL segment...`
even better if you use backticks around variable names